client.go

```diff
@@ -59,19 +59,30 @@ func NewClientWithTimeout(timeout time.Duration) *Client {
 }
 
 type federationTripper struct {
-	transport http.RoundTripper
+	transports map[string]http.RoundTripper
 }
 
 func newFederationTripper() *federationTripper {
-	// TODO: Verify ceritificates
-	return &federationTripper{
-		transport: &http.Transport{
+	tripper := new(federationTripper)
```

Comment:
I'm no go expert (as you will have gathered...) but is this more idiomatic than:

```
 &federationTripper{
    transports:  make(map[string]http.RoundTripper),
 }
```
?

Reply:
Not sure, I did it the way I'm used to without thinking much about it, but I'm under the impression that people usually either do one or the other.
Writing this comment made me look for discussions about which one may be more idiomatic, and I saw some people mention that `&Type{...}` is usually preferred in this kind of situation, so I'll change it to use that.